### PR TITLE
Upgradable staking smart contract

### PR DESCRIPTION
**Pull request by francisco-leal**

### CODE REVIEW

File: scripts/countEvents.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { ethers } from "hardhat";
```
Comment (francisco-leal) [inline scripts/countEvents.ts:1]: do we need this script?

File: test/contracts/TalentFactory.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ describe("TalentFactory", () => {
     [creator, minter, talent1, talent2] = await ethers.getSigners();
 
     TalentFactoryFactory = await ethers.getContractFactory("TalentFactory");
+    upgrades.silenceWarnings()
```
Comment (francisco-leal) [inline test/contracts/TalentFactory.ts:33]: should we revisit this? cc/ @verdefred

File: scripts/countEvents.ts (hunk 1 above)
Comment (RubenSousaDinis) [reply scripts/countEvents.ts:1]: Nope. Can be removed.

File: scripts/deployTalentToken.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { ethers, upgrades, waffle } from "hardhat";
+import * as FactoryArtifactV2 from "../artifacts/contracts/TalentFactoryV2.sol/TalentFactoryV2.json";
+import * as TalentTokenArtifactV2 from "../artifacts/contracts/test/TalentTokenV2.sol/TalentTokenV2.json";
+import { UpgradeableBeacon__factory } from "../typechain";
+
+const { exit } = process;
+const { deployContract } = waffle;
+
+async function main() {
```
Comment (RubenSousaDinis) [inline scripts/deployTalentToken.ts:9]: I think this is a duplicated script: Check: [scripts/changeProdMinters.ts](https://github.com/talentprotocol/contracts/pull/54/files#diff-5793189b2da10d3cccebd73f7ed368198e8496048a219fb37d9bb53b14db54d2)

File: test/contracts/TalentFactory.ts (hunk 2 above)
Comment (verdefred) [reply test/contracts/TalentFactory.ts:33]: yes, I'll try to spend some time to figure out more in depth why this was needed and try to remove it

**Review by RubenSousaDinis — APPROVED** [review]

**Pull request merged**